dissect/target/plugins/os/unix/shadow.py

```diff
@@ -128,3 +140,11 @@ def extract_crypt_details(shent: dict) -> dict:
         crypt["algo"] = algos[crypt["algo"]]
 
     return crypt
+
+
+def epoch_days_to_datetime(days: int) -> datetime:
+    """Convert a number representing the days since 1 January 1970 to a datetime object."""
+    if not isinstance(days, int):
+        raise ValueError("days argument should be an integer")
```

Comment:
If an invalid value is somehow passed now, the whole plugin will stop parsing possible remaining valid lines. Could you change the logic so this will be handled? The rest looks good!